src/app/SIPUserAgents/SIPUserAgent.cs

```diff
@@ -1407,6 +1407,22 @@ private Task<SocketError> ReinviteRequestFinalResponseReceived(SIPEndPoint local
                     MediaSession.SetRemoteDescription(SdpType.answer, SDP.ParseSDPDescription(sipResponse.Body));
                 }
             }
+            else if ((sipResponse.Status == SIPResponseStatusCodesEnum.ProxyAuthenticationRequired || sipResponse.Status == SIPResponseStatusCodesEnum.Unauthorised) && m_answerSipAccount != null)
+            {

```

Comment:
If the original call was initiated with a SIPClientUserAgent (i.e. initiated by the sipsorcery library end) then the re-INIVTE request won't be able to authenticate. Is that the intention?

Reply:
No the intention is for any user agent to be able to respond

Hmm I guess that the m_answerSipAccount could be better named/used. You're right that if the sip account isn't present, we will not re-send with authentication, but since this sip Account is supplied with the constructor, you could have it for any use case (initiated from this user agent or not). 

This brings another point though, that If a user places a call from any of the call method with a different set of username passwords than the ones in the SIP Account, then the authentication will be all over the place. 

I could add a check that if we initiated the call and we have a cached username/password, we use those for authentication and otherwise we can fallback on the sip account. 

I think this could get very confusing easily

Reply:
Yes, and also the same SIP request authentication logic is now being duplicated in multiple places. I'll need to have a think about a better way but if you do have any suggestions I'm happy to consider them.